Sources_v3/StreamChatUI/ChatChannel/ChatChannelNavigationBarListener.swift

```diff
@@ -11,17 +11,24 @@ class ChatChannelNavigationBarListener<ExtraData: ExtraDataTypes> {
     typealias NavbarData = (title: String?, subtitle: String?)
 
     let channelController: _ChatChannelController<ExtraData>
+    let namer: ChannelNamer
     var onDataChange: (NavbarData) -> Void = { _ in }
 
-    static func make(for channel: ChannelId, in client: _ChatClient<ExtraData>) -> ChatChannelNavigationBarListener {
+    static func make(
+        for channel: ChannelId,
+        in client: _ChatClient<ExtraData>,
+        using namer: ChannelNamer
+    ) -> ChatChannelNavigationBarListener {
         /// if we in channel room, channel will be here, but it always safe to fallback to group chat
         let isDirect = client.channelController(for: channel).channel?.isDirectMessageChannel ?? false
         return isDirect
-            ? DirectChatChannelNavigationBarListener(client: client, channel: channel)
-            : GroupChatChannelNavigationBarListener(client: client, channel: channel)
+            ? DirectChatChannelNavigationBarListener(client: client, channel: channel, namer: namer)
+            : GroupChatChannelNavigationBarListener(client: client, channel: channel, namer: namer)
     }
 
-    fileprivate init(client: _ChatClient<ExtraData>, channel: ChannelId) {
+    fileprivate init(client: _ChatClient<ExtraData>, channel: ChannelId, namer: ChannelNamer) {
```

Comment:
why not just `private`? IMHO `fileprivate` is almost unnecessary access control modifier. I don't see doing anything in any extension either. I would personally add some rule for not using `fileprivate`...

Reply:
Because subclasses needs to access this `init`